cpp/src/replace/replace.cu

```diff
@@ -13,31 +16,528 @@
+                     "values_to_replace and replacement_values size mismatch.");
+
+    if (0 == values_to_replace.size())
+    {
+      return std::make_unique<cudf::column>(input_col);
+    }
+
+
+    CUDF_EXPECTS(input_col.type() == values_to_replace.type() &&
+                 input_col.type() == replacement_values.type(),
+                 "Columns type mismatch.");
+    CUDF_EXPECTS(values_to_replace.nullable() == false,
+                 "Nulls are in values_to_replace column.");
+
+    std::unique_ptr<column> output;
+    if (input_col.nullable() || replacement_values.nullable()) {
+      output = cudf::experimental::allocate_like(input_col,
+                                                 cudf::experimental::mask_allocation_policy::ALWAYS,
+                                                 mr);
+    }
+    else
+      output = cudf::experimental::allocate_like(input_col,
+                                                 cudf::experimental::mask_allocation_policy::NEVER,
+                                                 mr);
+
+    cudf::mutable_column_view outputView = output->mutable_view();
+    cudf::experimental::type_dispatcher(input_col.type(),
+                                        replace_kernel_forwarder { },
+                                        input_col,
+                                        values_to_replace,
+                                        replacement_values,
+                                        outputView,
+                                        stream);
+
+    CHECK_STREAM(stream);
+    return output;
+  }
+
+} //end details
+namespace experimental {
+/* --------------------------------------------------------------------------*/
+/**
+ * @brief Replace elements from `input_col` according to the mapping `values_to_replace` to
+ *        `replacement_values`, that is, replace all `values_to_replace[i]` present in `input_col`
+ *        with `replacement_values[i]`.
+ *
+ * @param[in] col gdf_column with the data to be modified
+ * @param[in] values_to_replace gdf_column with the old values to be replaced
+ * @param[in] replacement_values gdf_column with the new values
+ *
+ * @returns output gdf_column with the modified data
+ */
+/* ----------------------------------------------------------------------------*/
+  std::unique_ptr<cudf::column> find_and_replace_all(cudf::column_view const& input_col,
+                                                     cudf::column_view const& values_to_replace,
+                                                     cudf::column_view const& replacement_values,
+                                                     rmm::mr::device_memory_resource* mr){
+    return cudf::detail::find_and_replace_all(input_col, values_to_replace, replacement_values, mr, 0);
+  }
+} //end experimental
+} //end cudf
+
+namespace{ //anonymous
+
+template <typename Type, typename iter>
+__global__
+void replace_nulls(cudf::size_type size,
```

Comment:
This kernel doesn't do anything to update the null mask or null count for the output column.

Reply:
It doesn't because it's currently not allowed to have nulls in the output (This was in the code before porting).

Reply:
Oh duh, this is the replace_nulls kernel. I got my wires crossed and thought I was looking at the replace all kernel.